code/modules/organs/limbs.dm

```diff
@@ -590,6 +610,29 @@ Note that amputating the affected organ does in fact remove the infection from t
 		limb_status |= LIMB_BLEEDING
 
 
+/datum/limb/proc/set_limb_destroyed(new_status = TRUE)
+	if(limb_status & LIMB_DESTROYED)
+		if(new_status != FALSE)
+			return
+		limb_status &= ~LIMB_DESTROYED
+		return new_status
+	if(new_status != TRUE)
+		return
+	limb_status |= LIMB_DESTROYED
+	return new_status
```

Comment:
Could this not be simplier like
```suggestion
/datum/limb/proc/set_limb_destroyed(new_status = TRUE)
	. = new_status // return the new state of the limb
	if((limb_status & LIMB_DESTROYED) == new_status) // limb is already the state we want
		return
	limb_status ^= LIMB_DESTROYED // toggle the limb destroyed
	return
```

Reply:
That would work as well, but the idea is to return `null` if no status change took place, so that procs that call it and update on change can ignore such calls.
It's a pattern I've been using on several places for update-on-demand statuses.

Reply:
Oh in which just remove the default return assignment, and `return new_status` on the last line